utilities/ismrmrd_hdf5_to_stream.cpp

```diff
@@ -0,0 +1,138 @@
+#include "ismrmrd/dataset.h"
+#include "ismrmrd/serialization.h"
+#include <boost/program_options.hpp>
+#include <iostream>
+#include <string>
+
+namespace po = boost::program_options;
+
+int main(int argc, char **argv) {
+
+    // Parse arguments using boost program options
+    po::options_description desc("Allowed options");
+
+    // Arguments
+    std::string input_file;
+    std::vector<std::string> image_series;
+    std::string groupname;
+
+    // clang-format off
+    desc.add_options()
+        ("help,h", "produce help message")
+        ("input,i", po::value<std::string>(&input_file)->required(),"input ISMRMRD HDF5 file")
+        ("group,g", po::value<std::string>(&groupname)->default_value("dataset"), "group name")
+        ("image-series,s", po::value<std::vector<std::string>>(&image_series)->multitoken(), "image series to extract");
+    // clang-format on
+
+    po::variables_map vm;
+    try {
+        po::store(po::parse_command_line(argc, argv, desc), vm);
+        po::notify(vm);
+    } catch (boost::wrapexcept<po::required_option> &e) {
+        std::cerr << "Error: " << e.what() << std::endl;
+        std::cerr << desc << std::endl;
+        return 1;
+    }
+
+    if (vm.count("help")) {
+        std::cerr << desc << "\n";
+        return 1;
+    }
+
+    ISMRMRD::Dataset d(input_file.c_str(), groupname.c_str(), false);
+    ISMRMRD::WritableStream ws(std::cout);
+    ISMRMRD::ProtocolSerializer serializer(ws);
+
+    try {
+        std::string xml;
+        d.readHeader(xml);
+        ISMRMRD::IsmrmrdHeader hdr;
+        ISMRMRD::deserialize(xml.c_str(), hdr);
+        serializer.serialize(hdr);
+    } catch (std::exception &e) {
+        if (image_series.empty()) {
+            // rethrow exception if image series not specified
```

Comment:
Why are we otherwise swallowing the exception?

Reply:
The try is on whether or not we can get a header and write it. If we are not streaming images, we have to have a header, so we will rethrow whatever happened. If we do have images, it is OK, we can continue.